jax/_src/scipy/special.py

```diff
@@ -2637,6 +2637,260 @@ def hyp1f1(a: ArrayLike, b: ArrayLike, x: ArrayLike) -> Array:
 )
 
 
+def _hyp2f1_terminal(a, b, c, x):
+  """
+  The Taylor series representation of the 2F1 hypergeometric function
+  terminates when either a or b is a non-positive integer. See Eq. 4.1 and
+  Taylor Series Method (a) from PEARSON, OLVER & PORTER 2014
+  https://doi.org/10.48550/arXiv.1407.7786
+  """
+  # Ensure that between a and b, the negative integer parameter with the greater
+  # absolute value - that still has a magnitude less than the absolute value of
+  # c if c is non-positive - is used for the upper limit in the loop.
+  eps = 5e-16
```

Comment:
Use
```suggestion
  eps = jnp.finfo(x.dtype).eps
```
as 5e-16 would be too small for float32, for instance.

Reply:
Ok no problem, I set the value back to `jnp.finfo(x.dtype).eps`, but see response to other comment for question